### ISSUE TITLE
Sbt needs convenience method to join together dynamic tasks.

### ISSUE THREAD

**Issue by jsuereth** [sbt/sbt#1503]
Right now, if you're pulling tasks from different `Initialize` contexts, there's no good way to join them.

This method should exist somewhere in our API:

``` scala
// Note this method is missing in sbt API AFAICT, and needs to exist.
// The type describes what we're doing:  Joining together a bunch of dynamic tasks
// into a new Task which will execute the underyling tasks in parallel and join
// the results into a Seq[T] that we can then use.
def join[T](tasks: Seq[Initialize[Task[T]]]): Initialize[Task[Seq[T]]] =
  Def.settingDyn {
    // First we join all the setting intializations into a single one with all the raw tasks.
    val zero: Initialize[Seq[Task[T]]] = Def.setting(Nil)
    val folded: Initialize[Seq[Task[T]]] = tasks.foldLeft(zero) { (acc, current) =>
      acc.zipWith(current) { case (taskSeq, task) => task +: taskSeq }
    }
    // Now we call the appropraite join task method.
    folded.apply { tasks: Seq[Task[T]] => tasks.join }
  }
```

But it's an open question where/how to expose it.

**Comment by jsuereth** [comment]
Note, it could be the method we really need is:

``` scala
def joinSettings[T](settings: Seq[Initialize[Seq[T]]]): Initialzie[Seq[T]] = ...
```

As this would then expose a method of joining the initialized tasks and we can then use the already exposed `Seq[Task[_]].join`.

**Comment by dwijnand** [comment]
I wonder, does Def.sequential solve this now?

**Comment by fommil** [comment]
:+1:

**Comment by fommil** [comment]
I actually need to go from `Seq[Task[T]]` => `Task[Seq[T]]`, not `Initialize` in sight.

**Comment by fommil** [comment]
t'was easy

```scala
      val work: Seq[Task[Unit]] = ...
      work.foldLeft(task(Nil: List[Unit])) { (acc, el) =>
        (acc, el).map { case (done, next) => next :: done }
      }
```

**Comment by fommil** [comment]
and if I want `Unit` back (the whole thing is a side effect) then

```
      work.foldLeft(task((): Unit)) { (acc, el) =>
        (acc, el).map { case (done, next) => () }
      }
```

**Comment by eed3si9n** [comment]
Hopefully `Def.sequential` was it.
